modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js

```diff
@@ -116,15 +116,22 @@ export function getValueDetailsFromItem(item, fieldName) {
 			if (property === 'LANG') {
 				const languageId = Liferay.ThemeDisplay.getLanguageId();
 				const BCP47LanguageId = Liferay.ThemeDisplay.getBCP47LanguageId();
+				const defaultInstanceLanguageId = Liferay.ThemeDisplay.getDefaultLanguageId();
 
 				if (navigatedValue[languageId]) {
 					formattedProperty = languageId;
 				}
 				else if (navigatedValue[BCP47LanguageId]) {
 					formattedProperty = BCP47LanguageId;
 				}
+				else if (navigatedValue[defaultInstanceLanguageId]) {
+					formattedProperty = defaultInstanceLanguageId;
+				}
+				else if (navigatedValue['en_US']) {
+					formattedProperty = 'en_US';
+				}
 				else {
-					formattedProperty = Liferay.ThemeDisplay.getDefaultLanguageId();
+					formattedProperty = Object.keys(navigatedValue)[0];
```

Comment:
Why are you using navigatedValue? @markocikos, have you ever seen this utility before? Your review would be wiser than mine

Reply:
A new day, a new finding of bizarre code in FDS :sweat_smile: 
- Why do we localize field value in frontend? This should probably be done in backend, just like for field label. 
- If we are implementing localization in frontend: Why on earth do have this obtuse `LANG` placeholder in an array, as an indicator if a field should be localized? We should have something like an explicit `localizeValue` flag.
- If we are implementing localization in frontend: Why don't we have a concept of default localization? This PR is setting English as default, which seems wrong.
- So many variable names in this util are poorly named.
- It's questionable if `valuePath` is needed. `rootPropertyName` would not be needed is field name had a sensible format.
- A loop that sets a value (`navigatedValue`) on first iteration and has it as a key variable that in second iteration. In addition, there is an unnecessary mutation of `navigatedValue` between initial assignment and first iteration. What? :scream: 

Overall, this the ugliest code I have seen in a really long time. Maybe ever. But, most of it is unrelated to this PR.

With just PR in mind:
- The root cause of the bug is in the fact that Object backend does not provide any localization besides `en_US`. The only thing we may be improving in FDS is how to fail more gracefully. In this PR, we are defaulting first to English, and then to the first item in the list of localizations. I would only default to the first item. We can merge this as a small improvement, that technically "fixes" the bug.
- IMO the whole localization of field values should be redesigned and reworked. Localized value should be set in backend and passed as a field value. Just like for field label, and just like for almost everything else in DXP.
- Is there a reason to implement frontend localization? Maybe some future plan to add language change feature on FDS, without page reload? If yes, then we should (1) rework the way fields labels are set, and (2) improve response data format with default localization for fields, explicit localize flag, and simplified handling on FDS side.

Both temporary and proper solutions are a product call @dsanz @pablo-agulla

Reply:
Hi @markocikos, I removed the logic "en_US".

Reply:
@locpham97 Thanks for making the change, but I don't think we can merge this PR. The reason is that it does not achieve the expected result in [the JIRA ticket](https://issues.liferay.com/browse/LPS-161653):
>  Expected Result: System Fields with label translated

With this PR, system fields are **not** translated. They are in English, where they should be in the language of the user.

To achieve the real translation, we need to take a different approach, see https://github.com/liferay-frontend/liferay-portal/pull/2748#discussion_r993319427. We can either:
1. Extend returned data to include translations of `item.label` in all languages, including default  - or -
2. Translate in backend and send `item.label` as a simple string.

Reply:
Very good questions @markocikos. 

> Is there a reason to implement frontend localization? Maybe some future plan to add language change feature on FDS, without page reload?

AFAIK, user-triggered language change in DXP UI requires a full page reload for the `I18nSerlvet` and friends to kick in, so I would suspect of any attempt to making it happen in a SPA scenario. A different thing would be to have this "inside" the FDS, so that it could filter the incoming data and show it in whatever language it wants. That would need access to the backend but technically, not a full page reload. I don't see any value  in this, just mentioning as example. 

> IMO the whole localization of field values should be redesigned and reworked

I agree that code is weird at best. I also agree that this deserves some rework but some DXP code is already using this pattern to transfer data to the FDS so the change must be handled hollistically, and of course in a separate pull. Unless we find a case that justifies it, sending all potential translations for a field is a waste of resources (cpu frontend side and more importantly, network traffic), so at the very least, the option of sending only the "current" language should probably be managed by the backend.

> The only thing we may be improving in FDS is how to fail more gracefully.

Here we shall only discuss about how FDS should behave when incoming data is missing in the current language. As you pointed out, I would have the component fail to the first available localization as a sub-optimal (and cheap) fix. However, @locpham97 please note this is not fixing the source problem (lack of proper localization) just hides it.

I totally agree with this approach, plus a tech-debt task that we'd discuss separately to rework this.